flow/diff_context.h

```diff
@@ -0,0 +1,235 @@
+#ifndef FLUTTER_FLOW_DIFF_CONTEXT_H_
+#define FLUTTER_FLOW_DIFF_CONTEXT_H_
+
+#include <map>
+#include <vector>
+#include "flutter/fml/logging.h"
+#include "flutter/fml/macros.h"
+#include "third_party/skia/include/core/SkMatrix.h"
+#include "third_party/skia/include/core/SkRect.h"
+
+namespace flutter {
+
+#ifndef NDEBUG
```

Comment:
I couldn't find this documented anywhere, but it might be useful to add in the PR description what needs to be done to build with diff context active. Which modes is it active, etc. Also unifying the logic of when this gets set to be in a gn file might make it easier to reason about.

Currently it looks like you want it active when `flutter_runtime_mode != "debug"` or something along those lines. Which can be done in the gn file.

Reply:
IIRC the idea right now is that diff context should only be enabled in debug build so that the tests run on CL. After this lands and it becomes part of repaint logic, there will be a runtime flag to enable this.